src/core/vdom.ts

```diff
@@ -1682,13 +1685,15 @@ export function renderer(renderer: () => RenderResult): Renderer {
 					const widgetMeta = widgetMetaMap.get(next.id);
 					if (widgetMeta) {
 						widgetMeta.dirty = true;
+						if (!widgetMeta.rendering && _idToWrapperMap.has(next.id)) {
```

Comment:
Only queue an invalidation when not rendering and the widget is still rendered in the page